File: ios/Sources/CapacitorCalendarPlugin/Models/Results/CheckAllPermissionsResult.swift
import Capacitor

struct CheckAllPermissionsResult: JSResult {
    private let statesDict: [String: String]

    init(statesDict: [String: String]) {
        self.statesDict = statesDict
    }

    func toJSON() -> JSObject {
        var result = JSObject()
        result["result"] = statesDict as JSObject
        return result
    }
}

File: android/src/main/java/dev/barooni/capacitor/calendar/models/results/CheckAllPermissionsResult.kt
package dev.barooni.capacitor.calendar.models.results

import com.getcapacitor.JSObject
import com.getcapacitor.PermissionState
import dev.barooni.capacitor.calendar.models.enums.CalendarPermissionScope
import dev.barooni.capacitor.calendar.models.templates.JSResult

data class CheckAllPermissionsResult(
    val permissionStates: Map<CalendarPermissionScope, PermissionState>,
) : JSResult {
    override fun toJSON(): JSObject {
        val result = JSObject()
        permissionStates.forEach { (scope, state) -> result.put(scope.value, state.toString()) }
        return result
    }
}
